coral-incremental/src/main/java/com/linkedin/coral/incremental/RelNodeCostEstimator.java

```diff
@@ -0,0 +1,237 @@
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.logical.LogicalJoin;
+import org.apache.calcite.rel.logical.LogicalProject;
+import org.apache.calcite.rel.logical.LogicalUnion;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rex.RexCall;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexNode;
+
+import static java.lang.Math.*;
+
+
+public class RelNodeCostEstimator {
+
+  class CostInfo {
+    // TODO: we may also need to add TableName field.
+    Double cost;
+    Double row;
+
+    public CostInfo(Double cost, Double row) {
+      this.cost = cost;
+      this.row = row;
+    }
+  }
+
+  class JoinKey {
+    String leftTableName;
+    String rightTableName;
+    String leftFieldName;
+    String rightFieldName;
+
+    public JoinKey(String leftTableName, String rightTableName, String leftFieldName, String rightFieldName) {
+      this.leftTableName = leftTableName;
+      this.rightTableName = rightTableName;
+      this.leftFieldName = leftFieldName;
+      this.rightFieldName = rightFieldName;
+    }
+  }
+
+  private Map<String, Double> stat = new HashMap<>();
+
+  private Map<String, Double> distinctStat = new HashMap<>();
+
+  public void setStat(Map<String, Double> stat) {
+    this.stat = stat;
+  }
+
+  public void setDistinctStat(Map<String, Double> distinctStat) {
+    this.distinctStat = distinctStat;
+  }
+
+  private Double IOCostParam = 1.0;
+
+  private Double shuffleCostParam = 1.0;
+
+  public void setIOCostParam(Double IOCostParam) {
+    this.IOCostParam = IOCostParam;
+  }
+
+  public void setShuffleCostParam(Double shuffleCostParam) {
+    this.shuffleCostParam = shuffleCostParam;
+  }
+
+  public void loadStatistic(String configPath) {
+    try {
+      String content = new String(Files.readAllBytes(Paths.get(configPath)));
+      // Parse JSON string to JsonObject
+      JsonObject jsonObject = new JsonParser().parse(content).getAsJsonObject();
+      // Iterate over each table in the JSON object
+      for (Map.Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+        String tableName = entry.getKey();
+        JsonObject tableObject = entry.getValue().getAsJsonObject();
+
+        // Extract row count
+        Double rowCount = tableObject.get("RowCount").getAsDouble();
+
+        // Extract distinct counts
+        JsonObject distinctCounts = tableObject.getAsJsonObject("DistinctCounts");
+
+        stat.put(tableName, rowCount);
+
+        // Iterate over distinct counts
+        for (Map.Entry<String, JsonElement> distinctEntry : distinctCounts.entrySet()) {
+          String columnName = distinctEntry.getKey();
+          Double distinctCount = distinctEntry.getValue().getAsDouble();
+
+          distinctStat.put(tableName + ":" + columnName, distinctCount);
+        }
+
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
Would be more graceful to handle this catch block by throwing an exception, can create a custom exception if need be.